File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
class Solution {
    public String[] solution(String[] strings, int n) {
        String[] answer = new String[strings.length];
        
        for (int i = 0; i < strings.length; i++) {
            for (int j = i + 1; j < strings.length; j++) {
                char c1 = strings[i].charAt(n);
                char c2 = strings[j].charAt(n);
                
                if (c1 > c2 || (c1 == c2 && strings[i].compareTo(strings[j]) > 0)) {
                    String temp = strings[i];
                    strings[i] = strings[j];
                    strings[j] = temp;
                }
            }
        }
        
        answer = strings;
        
        return answer;
    }
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string[] solution(string[] strings, int n) {
        Array.Sort(strings, (a, b) => {
            if (a[n] == b[n]) 
                return string.Compare(a, b);
            return a[n].CompareTo(b[n]);
        });
        
        /** Linq 풀이
        
        return strings
            .OrderBy(s => s[n])
            .ThenBy(s => s)
            .ToArray();
            
        **/
    
        return strings;
    }
}